userspace/libsinsp/test/sinsp_metrics.ut.cpp

```diff
@@ -0,0 +1,164 @@
+// SPDX-License-Identifier: Apache-2.0
+/*
+Copyright (C) 2024 The Falco Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+
+*/
+#ifdef __linux__
+
+#include <gtest/gtest.h>
+#include "sinsp_with_test_input.h"
+#include <libsinsp/test/helpers/threads_helpers.h>
+
+TEST_F(sinsp_with_test_input, sinsp_metrics_collector)
+{
+	m_inspector.set_sinsp_stats_v2_enabled();
+	// Extra call to verify that we don't fail
+	m_inspector.set_sinsp_stats_v2_enabled();
+	DEFAULT_TREE
+	auto evt = generate_random_event(p2_t1_tid);
+	ASSERT_EQ(get_field_as_string(evt, "proc.nthreads"), "3");
+
+	/* Snapshot current metrics and get the updated metrics_snapshot buffer */
+	uint32_t test_metrics_flags = (METRICS_V2_KERNEL_COUNTERS | METRICS_V2_LIBBPF_STATS | METRICS_V2_RESOURCE_UTILIZATION | METRICS_V2_STATE_COUNTERS);
+	bool convert_memory_to_mb = true;
+	std::unique_ptr<libsinsp::metrics::metrics_collector> metrics_collector = std::make_unique<libsinsp::metrics::metrics_collector>(&m_inspector, test_metrics_flags, convert_memory_to_mb);
```

Comment:
Nit: All the information needed to figure out what the line does is already on the RHS. You can keep it DRY, if you want.
```suggestion
	auto metrics_collector = std::make_unique<libsinsp::metrics::metrics_collector>(&m_inspector, test_metrics_flags, convert_memory_to_mb);
```